src/main/java/com/programmers/java/Calculator.java

```diff
@@ -0,0 +1,99 @@
+            char ch = postExp.charAt(index);
+            if (Character.isDigit(ch)) {
+                numberStack.push(new BigDecimal(tokenizer(postExp)));
+            } else if (isOperator(ch)) {
+                numberStack.push(performOperation(ch, numberStack.pop(), numberStack.pop()));
+            }
+            index++;
+        }
+        return numberStack.pop();
+    }
+
+    private static String toPostfix(String expression) throws Exception {
+        Stack<Character> operatorStack = new Stack<>();
+        StringBuilder intermediateBuilder = new StringBuilder();
+        while(index < expression.length()) {
+            char ch = expression.charAt(index);
+            if(Character.isDigit(ch)) {
+                intermediateBuilder.append(tokenizer(expression)+ " ");
+            } else if (isOperator(ch)) {
+                while(!operatorStack.isEmpty() && getPriority(operatorStack.peek()) >= getPriority(ch)) {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == '(') {
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == ')') {
+                while (operatorStack.peek() != '(') {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.pop();
+            } else {
+                throw new Exception();
+            }
+
+            index++;
+        }
+        while (!operatorStack.isEmpty()) {
+            intermediateBuilder.append(operatorStack.pop());
+        }
+        return intermediateBuilder.toString();
+    }
+
+    private static String tokenizer(String exp) {
+        StringBuilder decimalBuilder = new StringBuilder();
+        while (index < exp.length() && (Character.isDigit(exp.charAt(index)) || exp.charAt(index) == '.')) {
+            decimalBuilder.append(exp.charAt(index));
+            index++;
+        }
+        index--;
+        return decimalBuilder.toString();
+    }
+    private static BigDecimal performOperation(Character operator, BigDecimal popLast, BigDecimal popSecondLast) {
+        switch (operator) {
+            case '+':
+                return popSecondLast.add(popLast);
+            case '-':
+                return popSecondLast.subtract(popLast);
+            case '*':
+                return popSecondLast.multiply(popLast);
+            case '/':
+                return popSecondLast.divide(popLast);
+        }
+        return BigDecimal.ZERO;
+    }
+
+    public static boolean isOperator(char ch) {
+        return (ch == '+' || ch == '-' || ch == '*' || ch == '/');
+    }
+
+    public static int getPriority(char ch) {
+        return (ch == '+' || ch == '-') ? 1 : (ch == '*' || ch == '/') ? 2 : 0;
+    }
+
+    public static void showHistory() throws Exception {
+        if(history.size() == 0) throw new Exception();

```

Comment:
list의 유틸 클래스를 사용하면 가독성이 더 좋아집니다.

```suggestion
        if(history.isEmpty()) throw new Exception();
```

그리고 저장된 히스토리가 없는게 예외를 발생시킬만한 로직은 아닌것 같네요.
히스토리가 없으면 문구를 출력해주기만 하면 될것 같습니다.

Reply:
가급적 한줄짜리라도 컨벤션을 유지하면 좋을것 같은데, intellij에 코드 스타일 컨벤션을 셋팅하고 자동으로 활용되게 해두면 도움될것 같아요.

Reply:
코드 스타일 컨벤션도 intellij에서 정할수 있다니 편리한 기능들을 많이 제공해주는 것 같습니다. 적용해 볼게요!